main/com.spotify/index.js

```diff
@@ -1027,6 +1027,72 @@ module.exports = class SpotifyDevice extends Tp.BaseDevice {
         }
     }
 
+    async do_add_song_to_playlist({
+        song,
+        playlist
+    }) {
+        if (this._testMode())
+            return;
+        let playListURI = await this.findPlaylist(playlist);
+        playListURI = playListURI.substring(playListURI.indexOf("playlist:") + 9);
+        let data = {
+            "uris": [String(song)]
+        };
+        return this.add_uris_to_playlist(playListURI, data);
+    }
+
+    async do_add_this_song_to_playlist({
+        playlist
+    }) {
+        const currentlyPlaying = await this.get_get_currently_playing();
+        if (currentlyPlaying.length === 0)
+            throwError("no_song_error");
+        let playListURI = await this.findPlaylist(playlist);
+        playListURI = playListURI.substring(playListURI.indexOf("playlist:") + 9);
+        let song = String(currentlyPlaying[0]["song"]);
+        let data = {
+            "uris": [song]
+        };
+        return this.add_uris_to_playlist(playListURI, data);
+
+    }
+
+    async findPlaylist(name) {
+        const searchResults = await this.search(name, "playlist", 1);
+        if (!Object.prototype.hasOwnProperty.call(searchResults, 'playlists') || searchResults.playlists.total === 0) throwError('disallowed_action');
```

Comment:
We should have a better error code and error message to indicate that we got the playlist name wrong.